whelktool/scripts/cleanups/2019/09/remove-marc:nonfilingChars-with-empty-string-value/remove-marc:nonFilingChars.groovy

```diff
@@ -0,0 +1,29 @@
+/*
+ *
+ * Deletes the key-value pair marc:nonfilingChars = "X" when X equals " " (space)
+ * for auth, bib and hold records.
+ *
+ * Number of affected records from STG:
+ * select count(id) from lddb where data#>>'{@graph,1,hasTitle,0,marc:nonfilingChars}' = ' ';
+ * 1938
+ *
+ * See LXL-2598 for more info.
+ *
+ */
+
+PrintWriter failedIDs = getReportWriter("failed-to-delete-Ids")
+PrintWriter scheduledForChange = getReportWriter("scheduled-for-change")
+
+where = "data#>>'{@graph,1,hasTitle,0,marc:nonfilingChars}' = ' '"
+def absentNonfilingChars = ["marc:nonfilingChars": " "]
+
+selectBySqlWhere(where, silent: false) { docItem ->
+    Map hasTitle = docItem.doc.data["@graph"][1]["hasTitle"][0]
```

Comment:
I think there may be nonfilingChars in any Title item, not just the first?

Reply:
Yes, me and @sarwik discussed this. I'm reworking it.

Reply:
Ah, perfect; now I understand the "close without merge"! :)